src/App.vue

```diff
@@ -1,12 +1,116 @@
 <template>
-  <div id="nav">
+  <div v-if="currentUser && !showLogOutMessage" id="nav">
     <router-link to="/">Home</router-link> |
+    <router-link to="/play">Play</router-link> |
     <router-link to="/categories">Categories</router-link> |
-    <router-link to="/questions">Questions</router-link>
+    <router-link to="/questions">Questions</router-link> |
+
+    <div class="dropdown" style="display: inline-block">
+      <a
+        class="dropdown-toggle"
+        type="button"
+        id="dropdownMenu"
+        data-bs-toggle="dropdown"
+        aria-expanded="false"
+      >
+        {{ currentUser }}
+      </a>
+      <ul class="dropdown-menu" aria-labelledby="dropdownMenu2">
+        <li>
+          <a class="dropdown-item" type="button" href="#" @click="this.$router.push('/profile')">
+            My Profile
+          </a>
+        </li>
+        <li>
+          <button
+            class="dropdown-item"
+            type="button"
+            href="#"
+            style="color: red"
+            data-bs-toggle="modal"
+            data-bs-target="#staticBackdrop"
+          >
+            Log-Out
+          </button>
+        </li>
+      </ul>
+    </div>
+  </div>
+
+  <div
+    class="modal fade"
+    id="staticBackdrop"
+    data-bs-backdrop="static"
+    data-bs-keyboard="false"
+    tabindex="-1"
+    aria-hidden="true"
+  >
+    <div class="modal-dialog">
+      <div class="modal-content">
+        <div class="modal-body">
+          <br />
+          Are you sure want to log-out?
+        </div>
+        <div class="modal-footer justify-content-center">
+          <button
+            type="button"
+            class="btn btn-success"
+            data-bs-dismiss="modal"
+            @click="logOut"
+          >
+            Yes
+          </button>
+          <button type="button" class="btn btn-danger" data-bs-dismiss="modal">
+            No
+          </button>
+        </div>
+      </div>
+    </div>
   </div>
-  <router-view/>
+  <div v-if="showLogOutMessage" class="alert alert-danger" role="alert">
+    Successfully logged-out, have a nice day !
+  </div>
+
+  <router-view />
 </template>
 
+<script>
+export default {
+  name: 'App',
+  watch: {
+    $route() {
+      if (localStorage.getItem('currentUser')) {
+        try {
+          this.currentUser = JSON.parse(localStorage.getItem('currentUser'));
+        } catch (e) {
+          localStorage.removeItem('currentUser');
+        }
+      }
+    },
+  },
+  data() {
+    return {
+      currentUser: null,
+      showLogOutMessage: false,
+    };
+  },
+  methods: {
+    logOut() {
+      this.showLogOutMessage = true;
+      this.currentUser = '';
+      localStorage.setItem('currentUser', JSON.stringify(this.currentUser));
```

Comment:
why set it to empty here string and remove it on page change?

Reply:
We use 'currentUser' values to prevent unauthorized page transitions. (By entering a value like 'localhost:8081/categories' in the address bar when not logged in, unauthorized access to the 'categories' page is avoided by using '$router.back( )' function in 'if structures' when the pages are created.)

Example usage at 'Profile.vue':

97    if (!this.currentUser) {
98        this.$router.back()
99    }

Reply:
normally we'd manage this differently but this is good for now